modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/operations/util/OperationReply.java

```diff
@@ -686,38 +512,268 @@ private Item appendReplaceInfo(Operation operation) throws JSONException {
+
+                    AtomicTag atomicTagElement = new AtomicTag();
+                    List<AddTag> addConfig;
+                    try {
+                        addConfig = appendAddInfo(policy);
+                        atomicTagElement.setAdds(addConfig);
+                        atomicTagElement.setCommandId(operation.getId());
+
+                        sequenceElement.setAtomicTag(atomicTagElement);
+                    } catch (WindowsOperationException e) {
+                        throw new WindowsOperationException("Error occurred while generating operation payload.", e);
+                    }
+                }
+            }
+            if (!replaceItems.isEmpty()) {
+                sequenceElement.setReplaces(replaceItems);
+            }
+            return sequenceElement;
+
+        } else {
+            return null;
+        }
+    }
+
+    public List<Item> buildMonitorOperation(List<ProfileFeature> effectiveMonitoringFeature) {
+        List<Item> monitorItems = new ArrayList<>();
+        Operation monitorOperation;
+        for (ProfileFeature profileFeature : effectiveMonitoringFeature) {
+
+            if (profileFeature.getFeatureCode().equals(PluginConstants
+                    .OperationCodes.CAMERA)) {
+                String cameraStatus = PluginConstants
+                        .OperationCodes.CAMERA_STATUS;
+
+                monitorOperation = new Operation();
+                monitorOperation.setCode(cameraStatus);
+                Item item = appendGetInfo(monitorOperation);
+                monitorItems.add(item);
+            }
+            if (profileFeature.getFeatureCode().equals(PluginConstants
+                    .OperationCodes.ENCRYPT_STORAGE)) {
+                String encryptStorageStatus = PluginConstants
+                        .OperationCodes.ENCRYPT_STORAGE_STATUS;
+
+                monitorOperation = new Operation();
+                monitorOperation.setCode(encryptStorageStatus);
+                Item item = appendGetInfo(monitorOperation);
+                monitorItems.add(item);
+            }
+            if (profileFeature.getFeatureCode().equals(PluginConstants
+                    .OperationCodes.PASSCODE_POLICY)) {
+                String passcodeStatus = PluginConstants
+                        .OperationCodes.DEVICE_PASSWORD_STATUS;
+
+                monitorOperation = new Operation();
+                monitorOperation.setCode(passcodeStatus);
+                Item item = appendGetInfo(monitorOperation);
+                monitorItems.add(item);
+            }
+        }
+        return monitorItems;
+    }
+
+    public List<Item> buildDeleteInfo(Operation operation) {
+        List<Item> deleteItems = new ArrayList<>();
+        Item deleteItem = new Item();
+        Target target = new Target();
+        String operationCode = operation.getCode();
+        if (operation.getCode().equals(PluginConstants.OperationCodes.PASSCODE_POLICY)) {
```

Comment:
As a best practice, always compare the non-deterministic variable values against known constants.